pulp_python/app/models.py

```diff
@@ -63,6 +66,38 @@ def content_handler(self, path):
             name = path.parts[1]
         elif path.match("pypi/*/json"):
             name = path.parts[1]
+        elif len(path.parts) and path.parts[0] == "simple":
+            # Temporary fix for PublishedMetadata not being properly served from remote storage
+            # https://github.com/pulp/pulp_python/issues/413
+            if settings.DEFAULT_FILE_STORAGE != "pulpcore.app.models.storage.FileSystem":
+                try:
+                    publication = self.publication or Publication.objects.filter(
+                        repository_version=self.repository.latest_version()).latest("pulp_created")
+                except ObjectDoesNotExist:
+                    return None
+                rel_path = f"{path}/index.html"
+                try:
+                    ca = publication.published_artifact.select_related(
+                        "content_artifact",
+                        "content_artifact__artifact",
+                    ).get(relative_path=rel_path).content_artifact
+                except ObjectDoesNotExist:
+                    return None
+                file = ca.artifact.file
```

Comment:
Is there a guarantee that at this point in the execution, the content artifact has a downloaded `artifact` and not a remote artifact?

Reply:
Yes, the `ca.artifact` is actually a `published_artifact` so it is guaranteed to exist.

Reply:
Oh, you're right.  Could you rename `ca` to something that doesn't evoke `ContentArtifact`?

Reply:
I can, but it is a `ContentArtifact`, `PublishedArtifact` uses a `ContentArtifact` to store its file.

Reply:
Gotcha.  I didn't see the `.content_artifact` at the end of line 83